java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java

```diff
@@ -1028,4 +1036,330 @@ public void testRemoveSpecificAPIpatterns() throws IOException {
+    String newContent =
+        Files.readAllLines(Paths.get(srcProp))
+            .stream()
+            .map(
+                x ->
+                    x.replace(
+                        "\"methodProperties\": [",
+                        "\"methodProperties\": [" + methodPropertiesToAdd))
+            .map(
+                x ->
+                    x.replace(
+                        "\"annotations\": [", "\"annotations\": [" + annotationPropertiesToAdd))
+            .map(x -> x.replace("$flagName$", flagNameCamelCase))
+            .map(x -> x.replace("$notIsTreated$", String.valueOf(!isTreated)))
+            .collect(joining("\n"));
+    Path trgt = Paths.get(trgtProp);
+    Files.deleteIfExists(trgt);
+    Files.write(trgt, newContent.getBytes(UTF_8), StandardOpenOption.CREATE);
+  }
+
+  @Test
+  public void testMethodChainTreated() throws IOException {
+    String stale_flag = "stale_flag";
+    String isTreated = "true";
+    String srcProp = "config/properties.json";
+    transformAndCreateNewPropertyFile(
+        srcProp, trgtProp, stale_flag, Boolean.parseBoolean(isTreated));
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", stale_flag);
+    b.putFlag("Piranha:IsTreated", isTreated);
+    b.putFlag("Piranha:ArgumentIndexOptional", "true");
+    b.putFlag("Piranha:Config", trgtProp);
+
+    BugCheckerRefactoringTestHelper bcr =
+        BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+    bcr = bcr.addInput("Parameter.java").expectUnchanged();
+    bcr = bcr.addInput("BoolParameter.java").expectUnchanged();
+    bcr.addInputLines(
+            "SomeParam.java",
+            "package com.uber.piranha;",
+            "interface SomeParam  {",
+            " @BoolParam(key = \"other_flag\")",
+            " BoolParameter otherFlag();",
+            " @BoolParam(key = \"stale_flag\")",
+            " BoolParameter staleFlag();",
+            " static SomeParam create(Parameter cp){",
+            "  return null;",
+            " }",
+            "}")
+        .addOutputLines(
+            "SomeParam.java",
+            "package com.uber.piranha;",
+            "interface SomeParam  {",
+            " @BoolParam(key = \"other_flag\")",
+            " BoolParameter otherFlag();",
+            "",
+            "",
+            " static SomeParam create(Parameter cp){",
+            "  return null;",
+            " }",
+            "}")
+        .addInputLines(
+            "MethodChainTest.java",
+            "package com.uber.piranha;",
+            "public class MethodChainTest{",
+            " public static void foobar(Parameter cp){",
+            " SomeParam sp = SomeParam.create(cp);",
+            "  if(sp.staleFlag().getValue()){",
+            "    System.out.println();",
+            "   }",
+            "  if(!sp.staleFlag().getValue()){",
```

Comment:
can we also test with:
1. increased chain length
2. reversing the method invocations with what is given in properties.json
3. having a method that only matches partially the method invocations in the properties.json

a bit more emphasis on testing this out. also, it may be easier to test this with this [style](https://github.com/uber/piranha/blob/master/java/piranha/src/test/resources/com/uber/piranha/XPFlagCleanerPositiveCases.java) of testing where input and output files are compared. The number of cases for the input line and output lines is one too many and can be hard to follow.